azure-mgmt-eventhub/src/test/java/com/microsoft/azure/management/eventhub/EventHubTests.java

```diff
@@ -342,27 +342,27 @@ public void canConfigureEventHubDataCapturing() throws Exception {
 
         Creatable<StorageAccount> storageAccountCreatable = storageManager.storageAccounts()
                 .define(stgName)
-                .withRegion(REGION)
-                .withNewResourceGroup(RG_NAME)
-                .withSku(StorageAccountSkuType.STANDARD_LRS);
+                    .withRegion(REGION)
+                    .withNewResourceGroup(RG_NAME)
+                    .withSku(StorageAccountSkuType.STANDARD_LRS);
 
         Creatable<EventHubNamespace> namespaceCreatable = eventHubManager.namespaces()
                 .define(namespaceName)
-                .withRegion(REGION)
-                .withNewResourceGroup(RG_NAME);
+                    .withRegion(REGION)
+                    .withNewResourceGroup(RG_NAME);
 
         final String containerName1 = "eventsctr1";
 
         EventHub eventHub1 = eventHubManager.eventHubs()
                 .define(eventHubName1)
-                .withNewNamespace(namespaceCreatable)
-                .withNewStorageAccountForCapturedData(storageAccountCreatable, containerName1)
-                .withDataCaptureEnabled()
-                // Window config is optional if not set service will choose default for it2
-                //
-                .withDataCaptureWindowSizeInSeconds(120)
-                .withDataCaptureWindowSizeInMB(300)
-                .create();
+                    .withNewNamespace(namespaceCreatable)
+                    .withNewStorageAccountForCapturedData(storageAccountCreatable, containerName1)
+                    .withDataCaptureEnabled()
```

Comment:
this can be default/implicit and only provide an overwrite, .withDataCaptureDisable() - update will have them both though. I had a similar pattern for a particular SQL firewall rule (withAccessFromAzureServices())...

Reply:
This cannot be default because enabling this add additional cost and any thing that cause unintentional cost should be explicit in my opinion.

Reply:
but the intention is already there since it's the user's code to specify the data capturing properties. It's like creating a virtual machine and you will ask the user to specifically set the state of the machine as powered on.

Reply:
What I tried to provide is,  allow user to configure them but not necessarily enable it at the time of creation and whenever their app want to capture events enable it and once they have enough data to analyze it then disable it and so on. Irrespective of you turn it on or off event hub continue to work.

This is different from vm example you described right? Vm is a top level resource and it is implicit that it will be automatically turned on upon creation same goes with event hub as soon as it is created it is live for consumption. When we create such top-level or use a Creatable it is known to incur cost. Eventhub data capture is an additional config and which you can setup but setup won't cost you money but soon you decide to start using it you pay for it and you can stop paying it by Turing it off separately but not Turing off the parent resource.